llvm/lib/CodeGen/ExpandMemCmp.cpp

```diff
@@ -279,6 +304,7 @@ void MemCmpExpansion::createResultBlock() {
 
 MemCmpExpansion::LoadPair MemCmpExpansion::getLoadPair(Type *LoadSizeType,
                                                        bool NeedsBSwap,
+                                                       Type *BSwapSizeType,
```

Comment:
We always have either (`NeedsBSwap == false`, `BSwapSizeType==nullptr`) or (`NeedsBSwap == true`, `BSwapSizeType!=nullptr`). We could simplify that.